app/controllers/providers_controller.rb

```diff
@@ -95,6 +96,21 @@ def search
     redirect_to provider_path(provider_code)
   end
 
+  def training_providers_courses
+    @courses = Course.includes(:provider)
```

Comment:
We should change the name of this as they're not really 'courses' that the variable holds. `course_csv_rows` maybe? Also, you could probably just build the csv in the controller and return it as a string without the need for the template.

Reply:
done